seismicpro/src/gather/gather.py

```diff
@@ -1295,27 +1298,65 @@ def _plot_seismogram(self, ax, title, x_ticker, y_ticker, x_tick_src=None, y_tic
         add_colorbar(ax, img, colorbar, divider, y_ticker)
         self._finalize_plot(ax, title, divider, event_headers, top_header, x_ticker, y_ticker, x_tick_src, y_tick_src)
 
+    #pylint: disable=invalid-name
     def _plot_wiggle(self, ax, title, x_ticker, y_ticker, x_tick_src=None, y_tick_src="time", norm_tracewise=True,
-                     std=0.5, color="black", event_headers=None, top_header=None, **kwargs):
+                     std=0.5, event_headers=None, top_header=None, lw=None, alpha=None, color='k', **kwargs):
         """Plot the gather as an amplitude vs time plot for each trace."""
         # Make the axis divisible to further plot colorbar and header subplot
         divider = make_axes_locatable(ax)
 
-        color = to_list(color)
-        if len(color) == 1:
-            color = color * self.n_traces
-        elif len(color) != self.n_traces:
-            raise ValueError('The number of items in `color` must match the number of plotted traces')
-
-        y_coords = np.arange(self.n_samples)
         std_axis = 1 if norm_tracewise else None
         with warnings.catch_warnings():
             warnings.simplefilter("ignore", category=RuntimeWarning)
             traces = std * ((self.data - np.nanmean(self.data, axis=1, keepdims=True)) /
                             (np.nanstd(self.data, axis=std_axis, keepdims=True) + 1e-10))
-        for i, (trace, col) in enumerate(zip(traces, color)):
-            ax.plot(i + trace, y_coords, color=col, **kwargs)
-            ax.fill_betweenx(y_coords, i, i + trace, where=(trace > 0), color=col)
+
+        xy = np.argwhere(traces > 0)
+        # shift trace amplitudes according to the trace index in the gather
+        amps = traces + np.arange(traces.shape[0]).reshape(-1, 1)
+        positive_amps = amps[tuple(xy.T)]
+
+        # Estimate values for Polygons transperency and Lines width
+        axes_width_inches = ax.get_window_extent().transformed(ax.figure.dpi_scale_trans.inverted()).width
+
+        FINE_RATIO = 150 / 7.75 # N_TRACES / N_INCHES
+        FINE_ALPHA_MIN = 0.25 # value from [0, 1]
+
+        alpha, lw = [FINE_RATIO * (axes_width_inches / len(traces)) if val is None else val for val in [alpha, lw]]
+        alpha, lw = np.clip([alpha, lw], [FINE_ALPHA_MIN, 0], [1, 1])
+
+        # plot all the traces once, then hide transitions between adjacanet traces
+        amps[:, -1] = np.nan
+        ax.plot(amps.ravel(), np.tile(range(amps.shape[1]), amps.shape[0]), color=color, lw=lw, **kwargs)
+
+        # find indices of Polygons boarders
+        boarders = np.argwhere(np.diff(xy[:, 1]) != 1).flatten()
```

Comment:
```suggestion
        boarders = np.argwhere((np.diff(xy[:, 0]) == 1) & (np.diff(xy[:, 1]) != 1)).ravel()
```
1. Check for the trace id as well, you check may fail in strange rare cases
2. Switch to `ravel` to avoid copying